compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2674,6 +2674,78 @@ public void visit(BLangInvocation iExpr) {
+        }
+
+        BSymbol symbol = symResolver.lookupMainSpaceSymbolInPackage(errorConstructorExpr.pos, env,
+                names.fromString(pkgAlias), names.fromString(typeName));
+        if (symbol == symTable.notFoundSymbol || symbol.tag != SymTag.ERROR) {
+            dlog.error(errorConstructorExpr.pos, DiagnosticErrorCode.UNDEFINED_ERROR_TYPE_DESCRIPTOR, typeName);
+            resultType = symTable.semanticError;
+            return;
+        }
+        BErrorType errorConstructorType = (BErrorType) symbol.type;
+        if (this.expType == symTable.noType) {
+            this.expType = errorConstructorType;
+        }
+
+        if (!types.isAssignable(errorConstructorType, this.expType)) {
+            dlog.error(errorConstructorExpr.pos, DiagnosticErrorCode.INCOMPATIBLE_TYPES, this.expType,
+                    errorConstructorType);
+            resultType = symTable.semanticError;
+        }
+
+        if (errorConstructorExpr.positionalArgs.isEmpty()) {
+            resultType = symTable.semanticError;
+            return;
+        }
+        checkExpr(errorConstructorExpr.positionalArgs.get(0), this.env, symTable.stringType);
+        if (errorConstructorExpr.positionalArgs.size() > 1) {
+            checkExpr(errorConstructorExpr.positionalArgs.get(1), this.env, symTable.errorOrNilType);
+        }
+
+        BType detailType = errorConstructorType.detailType;
+        if (detailType.tag == TypeTags.MAP) {
+            BType errorDetailTypeConstraint = ((BMapType) detailType).constraint;
+            for (BLangNamedArgsExpression namedArgExpr: getProvidedErrorDetails(errorConstructorExpr)) {
+                // TODO : check the assignability of the type of named-arg to value:Clonable
+                checkExpr(namedArgExpr, env);
+                if (!types.isAssignable(namedArgExpr.expr.type, errorDetailTypeConstraint)) {
+                    dlog.error(namedArgExpr.pos, DiagnosticErrorCode.INVALID_ERROR_DETAIL_ARG_TYPE, namedArgExpr.name,
+                            errorDetailTypeConstraint, namedArgExpr.expr.type);
+                    resultType = symTable.semanticError;
+                }
+            }
+
+            if (resultType == symTable.semanticError) {
+                return;
+            }
+        } else if (detailType.tag == TypeTags.RECORD) {
+            BRecordType targetErrorDetailRec = (BRecordType) errorConstructorType.detailType;
+            BRecordType recordType = createErrorDetailRecordType(errorConstructorExpr, targetErrorDetailRec);
+            if (resultType == symTable.semanticError) {
+                return;
+            }
+
+            if (!types.isAssignable(recordType, targetErrorDetailRec)) {
+                reportErrorDetailMissmatchError(errorConstructorExpr, targetErrorDetailRec, recordType);
+                resultType = symTable.semanticError;
+                return;
+            }
+        } else {
+            resultType = symTable.semanticError;
```

Comment:
Shouldn't we return here? Or it gets overridden in L2746?

Reply:
Fixed